### PR TITLE
Supports PodSecurityContext and Gatling runner container securityContext

### PR DESCRIPTION
**Pull request by kane8n**
### Description

_Provide a description of what has been changed_

### Checklist

_Please check if applicable_

- [x] Tests have been added (if applicable, ie. when operator codes are added or modified)
- [x] Relevant docs have been added or modified (if applicable, ie. when new features are added or current features are modified)

### Relevant issue
https://github.com/st-tech/gatling-operator/issues/117

### operation check
created user and group in the Dockerfile of the sample scenario and confirmed that it works by setting the user/group created in the securityContext.

### CODE REVIEW

**Comment by annappropriate** [comment]
This is great, I would love to be able to specify the sysctls for the port range!

File: docs/api.md [hunk 1]
```diff
@@ -136,6 +134,8 @@ _Appears in:_
 | `tolerations` _[Toleration](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#toleration-v1-core) array_ | (Optional) Tolerations specification. |
 | `serviceAccountName` _string_ | (Required) ServiceAccountName specification. |
 | `volumes` _[Volume](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#volume-v1-core) array_ | (Optional) volumes specification. |
+| `securityContext` _[PodSecurityContext](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#podsecuritycontext-v1-core)_ | (Optional) SecurityContext specification. |
+| `runnerContainerSecurityContext` _[SecurityContext](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#securitycontext-v1-core)_ | (Optional) RunecontainerSecurityContext specifies the SecurityContext of the Gatling runner container. |
```
Comment (gold-kou) [inline docs/api.md:138]: typo

**Review by gold-kou — COMMENTED** [review]
Thanks for your PR.
Just nits.

File: config/samples/gatling-operator_v1alpha1_gatling01.yaml [hunk 2]
```diff
@@ -8,6 +8,13 @@ spec:
   notifyReport: false                                     # The flag of notifying gatling report
   cleanupAfterJobDone: true                               #  The flag of cleaning up gatling jobs resources after the job done
   podSpec:
+    securityContext:
+      sysctls:
+        - name: net.ipv4.ip_local_port_range
+          value: "1024 65535"
+    runnerContainerSecurityContext:
+      runAsUser: 1000
```
[the remark this answers is not in the document]
Comment (kane8n) [reply config/samples/gatling-operator_v1alpha1_gatling01.yaml:16]: @gold-kou 
This configuration is a sample to show that SecurityContext works, and the user/group specified here is created in the Dockerfile to make the sample work.

[the remark this answers is not in the document]
Comment (gold-kou) [reply config/samples/gatling-operator_v1alpha1_gatling01.yaml:16]: I know this is just a sample, however even if sample,  I just thought the value should be appropriate one.
However, as u said this sample works as expected. I checked the value in a gatling runner container.

```
gatling@gatling-sample01-runner-4ml82:/opt/gatling$ sysctl net.ipv4.ip_local_port_range
net.ipv4.ip_local_port_range = 1024	65535
```

Looks cool.

**Review by gold-kou — APPROVED** [review]
LGTM

**Review by itiB — APPROVED** [review]
I've added `procps` to `gatling/Dockerfile` in order to use the `ps` command and entered the Pod to confirm that the settings are being applied correctly. I was able to verify that both `sysctls` and `port_range` are set, so the code you've written seems to be working well! Excellent job.

<details>
<summary>Checked point</summary>

## Test simulation manifest

```
apiVersion: gatling-operator.tech.zozo.com/v1alpha1
kind: Gatling
metadata:
  name: gatling-sample01
spec:
  generateReport: false
  cleanupAfterJobDone: true
  podSpec:
    securityContext:
      runAsUser: 0
      runAsGroup: 0
      sysctls:
        - name: net.ipv4.ip_local_port_range
          value: "1024 65535"
    runnerContainerSecurityContext:
      runAsUser: 1000
      runAsGroup: 1000
    serviceAccountName: "gatling-operator-worker"
    gatlingImage: ghcr.io/st-tech/gatling:latest
  testScenarioSpec:
    parallelism: 3
    simulationClass: "MyBasicSimulation"
    env:
      - name: ENV
        value: "dev"
      - name: CONCURRENCY
        value: "1"
      - name: DURATION
        value: "60"
```

## Detail about created pod

```
❯ make kind-sample-deploy
Cluster already exists
cd gatling && docker build -t gatling:20240618-120429 .
...
gatling.gatling-operator.tech.zozo.com/gatling-sample01 created

support-security-context ❯ k exec -it gatling-sample01-runner-6b6t2 -- /bin/sh
Defaulted container "gatling-runner" out of: gatling-runner, gatling-waiter (init)
$ ps -eo pid,user,uid,comm
    PID USER       UID COMMAND
      1 gatling   1000 sh
     15 gatling   1000 gatling.sh
     19 gatling   1000 java
     85 gatling   1000 java
    127 gatling   1000 sh
    142 gatling   1000 ps
$ sysctl net.ipv4.ip_local_port_range
net.ipv4.ip_local_port_range = 1024     65535
$ 
```

</details>

**Pull request merged**